### ISSUE TITLE
Run observable outside of angular zone

### ISSUE THREAD

**Issue by jludvice** [redhat-ipaas/ipaas-ui#209]
Why? see this article https://christianliebel.com/2016/11/angular-2-protractor-timeout-heres-fix/

It breaks protractor tests.

**Comment by jludvice** [comment]
@gashcrumb I tried to use ngZone in app initializer but I don't know how to properly inject ngZone :(

With this commit I'm getting error in browser console:
```
Error: Uncaught (in promise): TypeError: Cannot read property 'runOutsideAngular' of undefined
TypeError: Cannot read property 'runOutsideAngular' of undefined
```

**Comment by gashcrumb** [comment]
Think you need to add ngZone to this list ->

https://github.com/redhat-ipaas/ipaas-ui/blob/master/src/app/app.module.ts#L158

You'll need to add an `NgZone` import to this line ->

https://github.com/redhat-ipaas/ipaas-ui/blob/master/src/app/app.module.ts#L2

like this example -> https://angular.io/docs/ts/latest/api/core/index/NgZone-class.html

and then this is where you actually inject `ngZone: NgZone` ->

https://github.com/redhat-ipaas/ipaas-ui/blob/master/src/app/app.module.ts#L25

**Comment by dsimansk** [comment]
Thanks @gashcrumb, I got it working locally. Josef will wrap it up and submit PR in a few minutes (no pressure :)).

**Comment by jludvice** [comment]
Hello @gashcrumb thanks for explanation. 
I was missing just this last step
```ts
  providers: [
    {
      provide: APP_INITIALIZER,
      useFactory: appInitializer,
      deps: [ConfigService, OAuthService, UserService, NgZone], //  <-- here
      multi: true,
    },
```

My original commit is gone (it was based on my branch with tests), this one is based on master branch.
Can you review if it works correctly ?

@jimmidyson @kahboom 
Just a note for the future - can we default to `ngZone.runOutsideAngular(()` in case we need timeout or observable?

Protractor waiting worked correctly for me with this fix.

**Comment by jludvice** [comment]
Resolved by #219

**Comment by kahboom** [comment]
Ty @jludvice !
